### ISSUE TITLE
Unable to install setfit in Kaggle Notebook

### ISSUE THREAD

**Issue by cassinthangam4996** [huggingface/setfit#294]
I was trying to install setfit in a new Kaggle notebook using GPU and it failed. It worked successfully for CPU. Then I looked into the closed issue #121 where they had mentioned the fix to uninstall tensorflow and then install setfit. I tried it but still I am getting the same error.

Here is the notebook : [https://www.kaggle.com/code/cassin4996/notebook28c45ee3f7](url)

<img width="720" alt="Screenshot 2023-02-01 at 11 03 11 PM" src="https://user-images.githubusercontent.com/22761661/216228880-665bcb52-dd5a-4027-b378-03f0b9060411.png">

**Comment by tomaarsen** [comment]
Hello!

It looks like despite all the red messages, SetFit and its dependencies were installed successfully. It seems that the removal of tensorflow simply made some other packages that rely on it understandably upset. I don't suspect this should be a problem, as long as you do not use these packages directly.

But I can't be certain, as your notebook seems to 404 for me.

- Tom Aarsen

**Comment by tomaarsen** [comment]
This seems to suggest that SetFit was installed. The ImportWarning error indicates that `datasets` considers the existing `pyarrow` version to be too old. `datasets` requires `pyarrow>=6.0.0`, and thus it uninstalls the existing `pyarrow` at `5.0.0` and installs the most recent version instead (`11.0.0`).
However, some of the dependencies that are preinstalled seem to complain about this, in particular `apache-beam`. Note that SetFit does not rely on `apache-beam`, `tensorflow`, `tensorflow-transform` nor `trx-bsl`.

I've tried installing SetFit on Kaggle just now, and I experienced no issues, and I think the main difference is that in my test notebook only had to install these modules:
```
Successfully installed datasets-2.9.0 evaluate-0.4.0 sentence-transformers-2.2.2 setfit-0.5.0
```
This differs your very first screenshot, which shows that you also had to install `pyarrow`. I think this is because my (newer) Kaggle environment already comes preloaded with `pyarrow>=6.0.0`, and so `datasets` won't try to install a more recent version.

See my test notebook here: https://www.kaggle.com/cubiedev/setfit-install-test

In conclusion, I believe you can fix your problem in two ways:
1. Using the latest environment:
    ![image](https://user-images.githubusercontent.com/37621491/216364856-db808ced-116f-4e76-b725-ecda302a7226.png)
2. Using `!pip install setfit pyarrow==6.0.0` (or some version between `6.0.0` inclusive and `8.0.0` exclusive). That way you'll satisfy both the `datasets` and `apache-beam` requirements.

- Tom Aarsen

**Comment by tomaarsen** [comment]
Ah, that is indeed the difference. I used the default accelerator, i.e. none. I suspect that most users of SetFit that rely on Kaggle notebooks would use it with a GPU accelerator, so a broken installation is problematic. I can reproduce the problem now.

It seems to be related to `datasets` still complaining that `pyarrow` is at version 5.0.0, despite version 11.0.0 being installed, due to a surprising value of `pyarrow.__version__`:
![image](https://user-images.githubusercontent.com/37621491/216464916-ac91a421-5dd3-4cde-8969-1792dbc095ec.png)

I'll try to look into it tomorrow.

- Tom Aarsen

**Comment by tomaarsen** [comment]
It seems that others have experienced issues updating pyarrow on Kaggle: https://www.kaggle.com/product-feedback/325765#2018166

**Comment by tomaarsen** [comment]
I understand your concern. The core of the issue is that Kaggle seems to use multiple Python environments, and so far I've only managed to install things in the environment that isn't actually used by the notebook cells:
![image](https://user-images.githubusercontent.com/37621491/218749250-295795e1-c193-426b-8988-0cb2f1da218e.png)

**Comment by Threepointone4** [comment]
@cassin-edwin. I think you need to restart runtime and try. 

Just to be clear. 
Step1: !pip install setfit

Step2: !pip install pyarrow==6.0.0 ( or any version you need)

**Step3: Restart Runtime** 

Step4: Start using setfit.
<img width="315" alt="image" src="https://user-images.githubusercontent.com/22583613/221872066-7423d9b1-5614-4521-9139-56da96bc7eaf.png">

Adding link for reference: [Kaggle Link](https://www.kaggle.com/code/vishwasgpai/setfit-install-test/edit).

Hope this Helped! Let me know if issue still there.

**Comment by tomaarsen** [comment]
@Threepointone4 Does that also work with a GPU Accelerator enabled?
![image](https://user-images.githubusercontent.com/37621491/221875884-f66e5290-3df3-46a7-81da-1892964e1368.png)

**Comment by Threepointone4** [comment]
@tomaarsen  I have tried with GPU. 
This is output of attached kaggle link and the configuration. 

P100
<img width="1221" alt="image" src="https://user-images.githubusercontent.com/22583613/221877299-c99dc667-65cf-4ad0-8f30-54ff99162932.png">

T4 x 2
<img width="1220" alt="image" src="https://user-images.githubusercontent.com/22583613/221877803-1a55a656-72f3-4111-8d2c-5030af42a367.png">

**Comment by tomaarsen** [comment]
Wonderful! Thank you for this.
